### PR TITLE
Refactor some Yokozuna tests to eliminate duplication

### PR DESCRIPTION
**Pull request by jvoegele**
Specifically, introduce the `gen_keys/1` and `assert_search/6` functions into the yokozuna_rt module and refactor tests to use them instead of copied and pasted code fragments.

### CODE REVIEW

File: src/yokozuna_rt.erl [hunk 1]
```diff
@@ -56,6 +58,14 @@
 host_entries(ClusterConnInfo) ->
     [riak_http(I) || {_,I} <- ClusterConnInfo].
 
+%% @doc Generate `SeqMax' keys. Yokozuna supports only UTF-8 compatible keys.
+-spec gen_keys(pos_integer()) -> list().
```
Comment (fadushin) [inline src/yokozuna_rt.erl]: I'm not an expert on xref, but are we allowed to say the return value is a list of binary?  [binary()], maybe?

File: src/yokozuna_rt.erl [hunk 2]
```diff
@@ -264,6 +274,20 @@ search_expect(solr, {Host, Port}, Index, Name0, Term0, Shards, Expect)
     {ok, "200", _, R} = ibrowse:send_req(URL, [], get, [], Opts),
     verify_count_http(Expect, R).
 
+assert_search(Pid, Cluster, Index, Search, SearchExpect, Params) ->
+    F = fun(_) ->
+                lager:info("Searching ~p and asserting it exists",
+                           [SearchExpect]),
+                {ok,{search_results,[{_Index,Fields}], _Score, Found}} =
+                    riakc_pb_socket:search(Pid, Index, Search, Params),
```
Comment (fadushin) [inline src/yokozuna_rt.erl]: I like the idea of abstracting this out, but we should probably use the pattern used in

https://github.com/basho/yokozuna/blob/2.0/riak_test/yz_pb.erl#L233

where we also pattern match against an empty set of results.  I don't know if we will get [] results in these tests, but I did hit them in the yz_pb tests.

File: src/yokozuna_rt.erl (hunk 1 above)
Comment (zeeshanlakhani) [inline src/yokozuna_rt.erl]: [binary()] would be correct.

File: src/yokozuna_rt.erl (hunk 2 above)
Comment (zeeshanlakhani) [inline src/yokozuna_rt.erl]: +1 @fadushin

**Comment by fadushin** [comment]
+1 32bd9cf

**Pull request merged**